mhs/outbound/outbound/request/synchronous/tests/test_handler.py

```diff
@@ -164,3 +168,54 @@ def test_post_with_no_body(self):
         self.assertEqual(response.code, 400)
         self.assertEqual(response.headers["Content-Type"], "text/plain")
         self.assertIn("Body missing from request", response.body.decode())
+
+    def test_handler_updates_store_for_sync_async(self):
+        expected_response = "Hello world!"
+        wdo = unittest.mock.MagicMock()
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(True)
+        result = test_utilities.awaitable((200, expected_response, wdo))
+
+        self.sync_async_workflow.handle_sync_async_outbound_message.return_value = result
+        self.config_manager.get_interaction_details.return_value = {'sync-async': True, 'workflow': WORKFLOW_NAME}
+
+        self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME}, body=REQUEST_BODY)
+        wdo.set_outbound_status.assert_called_with(wd.MessageStatus.OUTBOUND_SYNC_ASYNC_MESSAGE_SUCCESSFULLY_RESPONDED)
+
+    @patch('outbound.request.synchronous.handler.SynchronousHandler._write_response')
+    def test_handler_updates_store_for_sync_async_failure_response(self, write_mock):
+        write_mock.side_effect = Exception('Dam the connection was closed')
+        expected_response = "Hello world!"
+        wdo = unittest.mock.MagicMock()
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(True)
+        result = test_utilities.awaitable((200, expected_response, wdo))
+
+        self.sync_async_workflow.handle_sync_async_outbound_message.return_value = result
+        self.config_manager.get_interaction_details.return_value = {'sync-async': True, 'workflow': WORKFLOW_NAME}
+        self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME}, body=REQUEST_BODY)
+
+        wdo.set_outbound_status.assert_called_with(wd.MessageStatus.OUTBOUND_SYNC_ASYNC_MESSAGE_FAILED_TO_RESPOND)
+
+    def test_sync_async_workflow_invoked(self):
+        expected_response = "Hello world!"
+        wdo = unittest.mock.MagicMock()
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(True)
+        result = test_utilities.awaitable((200, expected_response, wdo))
+
+        self.sync_async_workflow.handle_sync_async_outbound_message.return_value = result
+        self.config_manager.get_interaction_details.return_value = {'sync-async': True, 'workflow': WORKFLOW_NAME}
+
+        self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME}, body=REQUEST_BODY)
+        self.sync_async_workflow.handle_sync_async_outbound_message.assert_called_once()
```

Comment:
I think it would probably be worth also verifying that the expected response is returned from the handler here (and in `test_sync_async_workflow_not_invoked` below), as that's the other part of its responsibilities.